### PR TITLE
Add reusable blocks documentation for remaining blocks

### LINKED ISSUE #40153: Create foundation for documentation of product editor blocks
## Description

We should create documentation for the blocks we created for use in the product editor. The initial focus should be on the blocks that have more reusability potential and are not bound to a fixed entity property, like:

- woocommerce/product-checkbox-field
- woocommerce/product-collapsible
- woocommerce/conditional
- woocommerce/product-taxonomy-field
- woocommerce/product-radio-field
- woocommerce/product-pricing-field

_(Please double check to see if some field that matches the criteria was left out of this list)_

Include screenshots on the documentation, so that the developer can easily see if the block can be used for their purpose.

## Acceptance criteria

- [ ] Define a format for the documentation of reusable product editor blocks
- [ ] Document all the reusable blocks in the defined format

### PR DESCRIPTION
**Pull request by nathanss**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This adds documentation for the remaining reusable blocks that we have, following the format that was defined in the `woocommerce/product-checkbox-field` block.

Closes #40153 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Read the docs and check how they look with the markdown rendered.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/blocks/pricing/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+- **Required:** `Yes`
+
+Property in which the price value is stored.
+
+### label
+
+- **Type:** `String`
+- **Required:** `Yes`
+
+Label that appears at the side of the price field.
```
Comment (octaedro) [inline packages/js/product-editor/src/blocks/pricing/README.md]: It's totally nitpicky, but in the example the label appears on top of the price field.

File: packages/js/product-editor/src/blocks/pricing/block.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
 	"keywords": [ "products", "price" ],
 	"textdomain": "default",
 	"attributes": {
-		"name": {
+		"property": {
```
Comment (octaedro) [inline packages/js/product-editor/src/blocks/pricing/block.json:11]: I think that [this 'name'](https://github.com/woocommerce/woocommerce/pull/40521/files#diff-0313866555230023b2954c4de2af62c3ff086977f8f604ec301ae35c622b744aR64) will always be `undefined`.

**Review by octaedro — COMMENTED** [review]
Great job @nathanss! The documentation looks good. I just left a couple of minor comments.

File: packages/js/product-editor/src/blocks/pricing/README.md (hunk 1 above)
Comment (nathanss) [reply packages/js/product-editor/src/blocks/pricing/README.md]: that's not nitpicky at all. Thanks for catching it.

**Review by octaedro — APPROVED** [review]
Thank you for the changes @nathanss! LGTM 🚀

**Pull request merged**